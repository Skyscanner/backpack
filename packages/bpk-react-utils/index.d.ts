/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/// <reference types="react" />
import wrapDisplayName from './src/wrapDisplayName';
import Portal from './src/Portal';
import TransitionInitialMount from './src/TransitionInitialMount';
import cssModules from './src/cssModules';
import deprecated from './src/deprecated';
import { isDeviceIphone, isDeviceIpad, isDeviceIos } from './src/deviceDetection';
import withDefaultProps from './src/withDefaultProps';
import isRTL from './src/isRTL';
import{ BpkDialogWrapper } from './src/BpkDialogWrapper/BpkDialogWrapper';
export { Portal, TransitionInitialMount, cssModules, deprecated, withDefaultProps, wrapDisplayName, isDeviceIphone, isDeviceIpad, isDeviceIos, isRTL, BpkDialogWrapper };
declare const _default: {
    Portal: typeof Portal;
    TransitionInitialMount: ({ appearActiveClassName, appearClassName, children, transitionTimeout, }: {
        appearClassName: string;
        appearActiveClassName: string;
        transitionTimeout: number;
        children: import("react").ReactNode;
    }) => JSX.Element;
    cssModules: (styles?: {
        [key: string]: any;
    }) => (...classNames: (string | number | boolean | {} | null | undefined)[]) => string;
    deprecated: any;
    withDefaultProps: (WrappedComponent: import("react").ComponentType<any>, defaultProps: {
        [rest: string]: any;
        className?: string | undefined;
    }) => {
        ({ children, className: innerClassName, ...rest }: {
            [rest: string]: any;
            children?: import("react").ReactNode;
            className?: string | null | undefined;
        }): JSX.Element;
        displayName: string;
    };
    wrapDisplayName: (Component: string | import("react").ComponentType<any> | null, hoc: string) => string;
    isDeviceIphone: () => boolean;
    isDeviceIpad: () => boolean;
    isDeviceIos: () => boolean;
    isRTL: () => boolean;
    BpkDialogWrapper: ({
      ariaLabelledby,
      children,
      closeOnEscPressed,
      closeOnScrimClick,
      dialogClassName,
      exiting,
      id,
      isOpen,
      onClose,
      timeout,
      transitionClassNames
    }: {
      ariaLabelledby: string;
      children: import("react").ReactNode;
      closeOnEscPressed?: boolean;
      closeOnScrimClick?: boolean;
      dialogClassName?: string;
      id: string | undefined;
      isOpen: boolean;
      onClose: (
        arg0?: Event | KeyboardEvent | MouseEvent | PointerEvent | TouchEvent,
        arg1?: {
          source: 'ESCAPE' | 'DOCUMENT_CLICK';
        },
    ) => void | null;
      exiting?: boolean;
      transitionClassNames?: {
        appear?: string,
        appearActive?: string,
        exit?: string
      };
      timeout?: {appear?: number, exit?: number};
    }) => JSX.Element;
};
export default _default;
