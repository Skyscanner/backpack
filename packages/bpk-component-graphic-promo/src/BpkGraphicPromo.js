/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow strict */

import PropTypes from 'prop-types';
import React from 'react';
import { cssModules } from 'bpk-react-utils';
import BpkText from 'bpk-component-text';
import BpkButton from 'bpk-component-button';
import BpkPanel from 'bpk-component-panel';

import STYLES from './BpkGraphicPromo.module.scss';

const getClassName = cssModules(STYLES);

export type Props = {
  className: ?string,
  kicker: ?string,
  headline: string,
  strapline: ?string,
  image: string,
  sponsorLabel: string,
  sponsorLogo: ?string,
  sponsorAltText: ?string,
  ctaText: string,
  ctaUrl: string,
  invertVertically: boolean,
  textAlign: 'start' | 'center' | 'end',
  textColor: 'white' | 'black',
};
const BpkGraphicPromo = ({
  className,
  ctaText,
  ctaUrl,
  headline,
  image,
  invertVertically,
  kicker,
  sponsorAltText,
  sponsorLabel,
  sponsorLogo,
  strapline,
  textAlign,
  textColor,
}: Props) => {
  const cardClasses = [getClassName('bpk-graphic-promo')];
  if (className) {
    cardClasses.push(className);
  }

  const containerClasses = [
    getClassName('bpk-graphic-promo__container'),
    getClassName(`bpk-graphic-promo__container--${textAlign}`),
  ];
  if (invertVertically) {
    containerClasses.push(
      getClassName('bpk-graphic-promo__container--inverted'),
    );
  }

  const getTextClasses = (baseClass: string) => {
    const classes = [getClassName(baseClass)];
    if (textAlign) {
      classes.push(getClassName(`${baseClass}--${textAlign}`));
    }

    return classes.join(' ');
  };

  return (
    <BpkPanel
      className={cardClasses.join(' ')}
      style={{ backgroundImage: `url(${image})`, color: textColor }}
      href={ctaUrl}
      padded={false}
    >
      <div className={containerClasses.join(' ')}>
        {sponsorLogo && (
          <div className={getTextClasses('bpk-graphic-promo__sponsor-content')}>
            <BpkText
              tagName="span"
              className={getClassName('bpk-graphic-promo__sponsor-label')}
            >
              {sponsorLabel}
            </BpkText>
            <img
              className={getClassName('bpk-graphic-promo__sponsor-logo')}
              alt={sponsorAltText}
              src={sponsorLogo}
            />
          </div>
        )}
        <div className={getTextClasses('bpk-graphic-promo__promo-content')}>
          {!sponsorLogo && kicker && (
            <BpkText
              tagName="span"
              className={getClassName('bpk-graphic-promo__kicker')}
            >
              {kicker}
            </BpkText>
          )}
          <BpkText
            tagName="h2"
            className={getClassName('bpk-graphic-promo__headline')}
          >
            {headline}
          </BpkText>
          {strapline && (
            <BpkText
              tagName="p"
              className={getClassName('bpk-graphic-promo__strapline')}
            >
              {strapline}
            </BpkText>
          )}
          <BpkButton
            className={getClassName('bpk-graphic-promo__cta')}
            href={ctaUrl}
            primary
          >
            {ctaText}
          </BpkButton>
        </div>
      </div>
    </BpkPanel>
  );
};

BpkGraphicPromo.propTypes = {
  className: PropTypes.string,
  kicker: PropTypes.string,
  headline: PropTypes.string.isRequired,
  strapline: PropTypes.string,
  image: PropTypes.string.isRequired,
  sponsorLabel: PropTypes.string,
  sponsorLogo: PropTypes.string,
  sponsorAltText: PropTypes.string,
  ctaText: PropTypes.string.isRequired,
  ctaUrl: PropTypes.string.isRequired,
  invertVertically: PropTypes.string.isRequired,
  textAlign: PropTypes.oneOf(['start', 'center', 'end']).isRequired,
  textColor: PropTypes.oneOf(['white', 'black']),
};

BpkGraphicPromo.defaultProps = {
  className: undefined,
  kicker: undefined,
  strapline: undefined,
  sponsorLabel: undefined,
  sponsorLogo: undefined,
  sponsorAltText: undefined,
  textColor: 'white',
};

export default BpkGraphicPromo;

// TODO: Things to do before submitting:
// 5) Confirm CTA button style.
// 6) Write accessibility tests.
// 7) Write unit tests.
// 8) Fix remaining ESLint issues before final commit.

// DONE
// 1) Translate the "Sponsored" text.
// 2) Set background color for when image fails to load.
// 3) Confirm background image aspect ratio / dimensions.
// 4) Confirm text colour.
