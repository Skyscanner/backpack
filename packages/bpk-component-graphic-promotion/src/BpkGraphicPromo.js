/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2022 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';

// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
import { cssModules } from '../../bpk-react-utils';
// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
import BpkText from '../../bpk-component-text';
// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
import BpkButton from '../../bpk-component-button';

import STYLES from './BpkGraphicPromo.module.scss';

const getClassName = cssModules(STYLES);

const ACCESSIBILITY_KEYS = {
  Enter: 13 /* Enter */,
  Space: 32 /* Space */,
};

const isAccessibilityClick = (event: React.KeyboardEvent<HTMLElement>) =>
  Object.keys(ACCESSIBILITY_KEYS).includes(event.key) ||
  Object.values(ACCESSIBILITY_KEYS).includes(event.keyCode || event.which);

export const TEXT_ALIGN = {
  start: 'start',
  center: 'center',
  end: 'end',
}; // as const; // Temp disabling TS types due to non TS projects compatibility.

export type Props = {
  className?: string | null,
  contentId?: string | null,
  tagline?: string | null,
  headline: string,
  subheading?: string | null,
  sponsor?: {
    label: string,
    logo: string,
    altText: string,
  } | null,
  buttonText: string,
  onClick: () => void,
  invertVertically?: boolean,
  textAlign: $Keys<typeof TEXT_ALIGN>, // typeof TEXT_ALIGN[keyof typeof TEXT_ALIGN]; // Temp disabling TS types due to non TS projects compatibility.
  style?: {},
};

const constructAriaLabel = ({
  buttonText,
  headline,
  sponsor,
  subheading,
  tagline,
}: Pick<
  Props,
  'buttonText' | 'headline' | 'sponsor' | 'subheading' | 'tagline',
>) => {
  const text: string[] = [];
  const addText = (value?: string | null) => value && text.push(value);

  if (sponsor) {
    addText(sponsor.label);
    addText(sponsor.altText);
  } else {
    addText(tagline);
  }

  addText(headline);
  addText(subheading);
  addText(buttonText);

  return text.join('. ');
};

const BpkGraphicPromo = ({
  buttonText,
  className = null,
  contentId,
  headline,
  invertVertically = false,
  onClick,
  sponsor = null,
  style = {},
  subheading = null,
  tagline = null,
  textAlign,
}: Props) => {
  // FIXME: Use useCallback() here when React is updated.
  const onClickWrapper = (event: React.MouseEvent<HTMLElement>) => {
    event.stopPropagation();
    onClick();
  };
  const onKeyWrapper = (event: React.KeyboardEvent<HTMLElement>) =>
    isAccessibilityClick(event) && onClick();

  const cardClasses = getClassName('bpk-card', 'bpk-graphic-promo', className);
  const containerClasses = getClassName(
    'bpk-graphic-promo__container',
    `bpk-graphic-promo__container--${textAlign}`,
    invertVertically && 'bpk-graphic-promo__container--inverted',
  );

  const getTextClasses = (baseClass: string) =>
    getClassName(baseClass, `${baseClass}--${textAlign}`);

  return (
    // The card appears as a single component for the screen reader; its children are hidden. The card handles mouse
    // clicks and key presses (Enter/Space) for the whole component, as described here:
    // https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets
    <div
      id={contentId || ''}
      className={cardClasses}
      style={style}
      role="link"
      aria-label={constructAriaLabel({
        buttonText,
        headline,
        sponsor,
        subheading,
        tagline,
      })}
      tabIndex={0}
      onClick={onClickWrapper}
      onKeyDown={onKeyWrapper}
    >
      <div
        id={(contentId && `${contentId}__content`) || ''}
        className={containerClasses}
        aria-hidden
      >
        <div className={getTextClasses('bpk-graphic-promo__sponsor-content')}>
          {sponsor && (
            <>
              <BpkText
                tagName="span"
                className={getClassName('bpk-graphic-promo__sponsor-label')}
              >
                {sponsor.label}
              </BpkText>
              <img
                className={getClassName('bpk-graphic-promo__sponsor-logo')}
                alt={sponsor.altText}
                src={sponsor.logo}
              />
            </>
          )}
        </div>
        <div className={getTextClasses('bpk-graphic-promo__promo-content')}>
          {!sponsor && tagline && (
            <BpkText
              tagName="span"
              className={getClassName('bpk-graphic-promo__tagline')}
            >
              {tagline}
            </BpkText>
          )}
          <BpkText
            tagName="h2"
            className={getClassName('bpk-graphic-promo__headline')}
          >
            {headline}
          </BpkText>
          {subheading && (
            <BpkText
              tagName="p"
              className={getClassName('bpk-graphic-promo__subheading')}
            >
              {subheading}
            </BpkText>
          )}
          <BpkButton
            primaryOnDark
            className={getClassName('bpk-graphic-promo__cta')}
            onClick={onClickWrapper}
            tabIndex={-1} /* button is not focusable for accessibility */
          >
            {buttonText}
          </BpkButton>
        </div>
      </div>
    </div>
  );
};

BpkGraphicPromo.defaultProps = {
  className: null,
  contentId: null,
  tagline: null,
  subheading: null,
  sponsor: null,
  invertVertically: false,
  style: {},
};

export default BpkGraphicPromo;
