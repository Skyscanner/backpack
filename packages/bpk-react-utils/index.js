/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow strict */

import wrapDisplayName from './src/wrapDisplayName';
import Portal from './src/Portal';
import TransitionInitialMount from './src/TransitionInitialMount';
import cssModules from './src/cssModules';
import deprecated from './src/deprecated';
import {
  isDeviceIphone,
  isDeviceIpad,
  isDeviceIos,
} from './src/deviceDetection';
import withDefaultProps from './src/withDefaultProps';
import isRTL from './src/isRTL';

export {
  Portal,
  TransitionInitialMount,
  cssModules,
  deprecated,
  withDefaultProps,
  wrapDisplayName,
  isDeviceIphone,
  isDeviceIpad,
  isDeviceIos,
  isRTL,
};
export default {
  Portal,
  TransitionInitialMount,
  cssModules,
  deprecated,
  withDefaultProps,
  wrapDisplayName,
  isDeviceIphone,
  isDeviceIpad,
  isDeviceIos,
  isRTL,
};
