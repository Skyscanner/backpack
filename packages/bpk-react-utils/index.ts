/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import{ BpkDialogWrapper } from './src/BpkDialogWrapper/BpkDialogWrapper';
import Portal from './src/Portal';
import TransitionInitialMount from './src/TransitionInitialMount';
import cssModules from './src/cssModules';
// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
import deprecated from './src/deprecated';
import {
  isDeviceIphone,
  isDeviceIpad,
  isDeviceIos,
} from './src/deviceDetection';
import isRTL from './src/isRTL';
import withDefaultProps from './src/withDefaultProps';
import wrapDisplayName from './src/wrapDisplayName';

export {
  Portal,
  TransitionInitialMount,
  cssModules,
  deprecated,
  withDefaultProps,
  wrapDisplayName,
  isDeviceIphone,
  isDeviceIpad,
  isDeviceIos,
  isRTL,
  BpkDialogWrapper
};
export default {
  Portal,
  TransitionInitialMount,
  cssModules,
  deprecated,
  withDefaultProps,
  wrapDisplayName,
  isDeviceIphone,
  isDeviceIpad,
  isDeviceIos,
  isRTL,
  BpkDialogWrapper
};
