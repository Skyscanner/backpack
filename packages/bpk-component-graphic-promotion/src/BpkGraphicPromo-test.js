/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2022 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { fireEvent, render } from '@testing-library/react';

// @ts-expect-error Untyped import. Not added as a stub declaration so consumers still have the suppression in their node_modules when shipped untranspiled.
import { cssModules } from '../../bpk-react-utils';

import BpkGraphicPromo, { TEXT_ALIGN } from './BpkGraphicPromo';
import type { Props } from './BpkGraphicPromo';
import STYLES from './BpkGraphicPromo.module.scss';

const getClassName = cssModules(STYLES);

const props: Props = {
  buttonText: 'Learn more',
  headline: 'Ride your wave',
  invertVertically: false,
  onClick: jest.fn(),
  sponsor: {
    label: 'Sponsored',
    logo: 'path/to/logo.png',
    altText: 'Airline Name',
  },
  subheading:
    'Portugal and 6 more countries have just been added to the UK travel green list',
  tagline: 'Tagline',
  textAlign: TEXT_ALIGN.start,
  contentId: 'mock-content-id',
};

describe('BpkGraphicPromo', () => {
  beforeEach(() => {
    // Temp disabling TS types due to non TS projects compatibility.
    // (props.onClick as jest.MockedFunction<Props['onClick']>).mockReset();
    props.onClick.mockReset();
  });

  it('should render correctly with all properties set', () => {
    const { asFragment } = render(<BpkGraphicPromo {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should support custom class names', () => {
    const { asFragment } = render(
      <BpkGraphicPromo className="custom-classname" {...props} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly when centre aligned', () => {
    const customProps = { ...props, textAlign: TEXT_ALIGN.center };
    const { asFragment } = render(<BpkGraphicPromo {...customProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly when right aligned', () => {
    const customProps = { ...props, textAlign: TEXT_ALIGN.end };
    const { asFragment } = render(<BpkGraphicPromo {...customProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly when inverted portrait', () => {
    const customProps = { ...props, invertVertically: true };
    const { asFragment } = render(<BpkGraphicPromo {...customProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should not display tagline or subheading when not provided', () => {
    const customProps = { ...props };
    delete customProps.tagline;
    delete customProps.subheading;
    const { asFragment } = render(<BpkGraphicPromo {...customProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display the tagline and no sponsor for non-sponsored ad', () => {
    const customProps = { ...props };
    delete customProps.sponsor;
    const { asFragment } = render(<BpkGraphicPromo {...customProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should redirect us to link when card is clicked', () => {
    render(<BpkGraphicPromo {...props} />);

    const graphicPromo = document.getElementsByClassName(
      getClassName('bpk-graphic-promo'),
    )[0];
    fireEvent.click(graphicPromo);

    expect(props.onClick).toHaveBeenCalledTimes(1);
  });

  it('should redirect us to link when button is clicked', () => {
    render(<BpkGraphicPromo {...props} />);

    const graphicPromoCTA = document.getElementsByClassName(
      getClassName('bpk-graphic-promo__cta'),
    )[0];
    fireEvent.click(graphicPromoCTA);

    expect(props.onClick).toHaveBeenCalledTimes(1);
  });

  it('should render as expected without a content ID', () => {
    const { asFragment } = render(
      <BpkGraphicPromo {...props} contentId={null} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
