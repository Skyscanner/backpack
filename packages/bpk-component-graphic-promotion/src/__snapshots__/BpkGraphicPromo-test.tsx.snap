// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkGraphicPromo should display the tagline and no sponsor for non-sponsored ad 1`] = `
<DocumentFragment>
  <div
    aria-label="Tagline. Ride your wave. Portugal and 6 more countries have just been added to the UK travel green list. Learn more"
    class="bpk-card bpk-graphic-promo"
    id="mock-content-id"
    role="link"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="bpk-graphic-promo__container bpk-graphic-promo__container--start"
      id="mock-content-id__content"
    >
      <div
        class="bpk-graphic-promo__sponsor-content bpk-graphic-promo__sponsor-content--start"
      />
      <div
        class="bpk-graphic-promo__promo-content bpk-graphic-promo__promo-content--start"
      >
        <span
          class="bpk-text bpk-text--body-default bpk-graphic-promo__tagline"
        >
          Tagline
        </span>
        <h2
          class="bpk-text bpk-text--body-default bpk-graphic-promo__headline"
        >
          Ride your wave
        </h2>
        <p
          class="bpk-text bpk-text--body-default bpk-graphic-promo__subheading"
        >
          Portugal and 6 more countries have just been added to the UK travel green list
        </p>
        <button
          class="bpk-button bpk-button--primary-on-dark bpk-graphic-promo__cta"
          tabindex="-1"
          type="button"
        >
          Learn more
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BpkGraphicPromo should not display tagline or subheading when not provided 1`] = `
<DocumentFragment>
  <div
    aria-label="Sponsored. Airline Name. Ride your wave. Learn more"
    class="bpk-card bpk-graphic-promo"
    id="mock-content-id"
    role="link"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="bpk-graphic-promo__container bpk-graphic-promo__container--start"
      id="mock-content-id__content"
    >
      <div
        class="bpk-graphic-promo__sponsor-content bpk-graphic-promo__sponsor-content--start"
      >
        <span
          class="bpk-text bpk-text--body-default bpk-graphic-promo__sponsor-label"
        >
          Sponsored
        </span>
        <img
          alt="Airline Name"
          class="bpk-graphic-promo__sponsor-logo"
          src="path/to/logo.png"
        />
      </div>
      <div
        class="bpk-graphic-promo__promo-content bpk-graphic-promo__promo-content--start"
      >
        <h2
          class="bpk-text bpk-text--body-default bpk-graphic-promo__headline"
        >
          Ride your wave
        </h2>
        <button
          class="bpk-button bpk-button--primary-on-dark bpk-graphic-promo__cta"
          tabindex="-1"
          type="button"
        >
          Learn more
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BpkGraphicPromo should render as expected without a content ID 1`] = `
<DocumentFragment>
  <div
    aria-label="Sponsored. Airline Name. Ride your wave. Portugal and 6 more countries have just been added to the UK travel green list. Learn more"
    class="bpk-card bpk-graphic-promo"
    id=""
    role="link"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="bpk-graphic-promo__container bpk-graphic-promo__container--start"
      id=""
    >
      <div
        class="bpk-graphic-promo__sponsor-content bpk-graphic-promo__sponsor-content--start"
      >
        <span
          class="bpk-text bpk-text--body-default bpk-graphic-promo__sponsor-label"
        >
          Sponsored
        </span>
        <img
          alt="Airline Name"
          class="bpk-graphic-promo__sponsor-logo"
          src="path/to/logo.png"
        />
      </div>
      <div
        class="bpk-graphic-promo__promo-content bpk-graphic-promo__promo-content--start"
      >
        <h2
          class="bpk-text bpk-text--body-default bpk-graphic-promo__headline"
        >
          Ride your wave
        </h2>
        <p
          class="bpk-text bpk-text--body-default bpk-graphic-promo__subheading"
        >
          Portugal and 6 more countries have just been added to the UK travel green list
        </p>
        <button
          class="bpk-button bpk-button--primary-on-dark bpk-graphic-promo__cta"
          tabindex="-1"
          type="button"
        >
          Learn more
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BpkGraphicPromo should render correctly when centre aligned 1`] = `
<DocumentFragment>
  <div
    aria-label="Sponsored. Airline Name. Ride your wave. Portugal and 6 more countries have just been added to the UK travel green list. Learn more"
    class="bpk-card bpk-graphic-promo"
    id="mock-content-id"
    role="link"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="bpk-graphic-promo__container bpk-graphic-promo__container--center"
      id="mock-content-id__content"
    >
      <div
        class="bpk-graphic-promo__sponsor-content bpk-graphic-promo__sponsor-content--center"
      >
        <span
          class="bpk-text bpk-text--body-default bpk-graphic-promo__sponsor-label"
        >
          Sponsored
        </span>
        <img
          alt="Airline Name"
          class="bpk-graphic-promo__sponsor-logo"
          src="path/to/logo.png"
        />
      </div>
      <div
        class="bpk-graphic-promo__promo-content bpk-graphic-promo__promo-content--center"
      >
        <h2
          class="bpk-text bpk-text--body-default bpk-graphic-promo__headline"
        >
          Ride your wave
        </h2>
        <p
          class="bpk-text bpk-text--body-default bpk-graphic-promo__subheading"
        >
          Portugal and 6 more countries have just been added to the UK travel green list
        </p>
        <button
          class="bpk-button bpk-button--primary-on-dark bpk-graphic-promo__cta"
          tabindex="-1"
          type="button"
        >
          Learn more
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BpkGraphicPromo should render correctly when inverted portrait 1`] = `
<DocumentFragment>
  <div
    aria-label="Sponsored. Airline Name. Ride your wave. Portugal and 6 more countries have just been added to the UK travel green list. Learn more"
    class="bpk-card bpk-graphic-promo"
    id="mock-content-id"
    role="link"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="bpk-graphic-promo__container bpk-graphic-promo__container--start bpk-graphic-promo__container--inverted"
      id="mock-content-id__content"
    >
      <div
        class="bpk-graphic-promo__sponsor-content bpk-graphic-promo__sponsor-content--start"
      >
        <span
          class="bpk-text bpk-text--body-default bpk-graphic-promo__sponsor-label"
        >
          Sponsored
        </span>
        <img
          alt="Airline Name"
          class="bpk-graphic-promo__sponsor-logo"
          src="path/to/logo.png"
        />
      </div>
      <div
        class="bpk-graphic-promo__promo-content bpk-graphic-promo__promo-content--start"
      >
        <h2
          class="bpk-text bpk-text--body-default bpk-graphic-promo__headline"
        >
          Ride your wave
        </h2>
        <p
          class="bpk-text bpk-text--body-default bpk-graphic-promo__subheading"
        >
          Portugal and 6 more countries have just been added to the UK travel green list
        </p>
        <button
          class="bpk-button bpk-button--primary-on-dark bpk-graphic-promo__cta"
          tabindex="-1"
          type="button"
        >
          Learn more
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BpkGraphicPromo should render correctly when right aligned 1`] = `
<DocumentFragment>
  <div
    aria-label="Sponsored. Airline Name. Ride your wave. Portugal and 6 more countries have just been added to the UK travel green list. Learn more"
    class="bpk-card bpk-graphic-promo"
    id="mock-content-id"
    role="link"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="bpk-graphic-promo__container bpk-graphic-promo__container--end"
      id="mock-content-id__content"
    >
      <div
        class="bpk-graphic-promo__sponsor-content bpk-graphic-promo__sponsor-content--end"
      >
        <span
          class="bpk-text bpk-text--body-default bpk-graphic-promo__sponsor-label"
        >
          Sponsored
        </span>
        <img
          alt="Airline Name"
          class="bpk-graphic-promo__sponsor-logo"
          src="path/to/logo.png"
        />
      </div>
      <div
        class="bpk-graphic-promo__promo-content bpk-graphic-promo__promo-content--end"
      >
        <h2
          class="bpk-text bpk-text--body-default bpk-graphic-promo__headline"
        >
          Ride your wave
        </h2>
        <p
          class="bpk-text bpk-text--body-default bpk-graphic-promo__subheading"
        >
          Portugal and 6 more countries have just been added to the UK travel green list
        </p>
        <button
          class="bpk-button bpk-button--primary-on-dark bpk-graphic-promo__cta"
          tabindex="-1"
          type="button"
        >
          Learn more
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BpkGraphicPromo should render correctly with all properties set 1`] = `
<DocumentFragment>
  <div
    aria-label="Sponsored. Airline Name. Ride your wave. Portugal and 6 more countries have just been added to the UK travel green list. Learn more"
    class="bpk-card bpk-graphic-promo"
    id="mock-content-id"
    role="link"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="bpk-graphic-promo__container bpk-graphic-promo__container--start"
      id="mock-content-id__content"
    >
      <div
        class="bpk-graphic-promo__sponsor-content bpk-graphic-promo__sponsor-content--start"
      >
        <span
          class="bpk-text bpk-text--body-default bpk-graphic-promo__sponsor-label"
        >
          Sponsored
        </span>
        <img
          alt="Airline Name"
          class="bpk-graphic-promo__sponsor-logo"
          src="path/to/logo.png"
        />
      </div>
      <div
        class="bpk-graphic-promo__promo-content bpk-graphic-promo__promo-content--start"
      >
        <h2
          class="bpk-text bpk-text--body-default bpk-graphic-promo__headline"
        >
          Ride your wave
        </h2>
        <p
          class="bpk-text bpk-text--body-default bpk-graphic-promo__subheading"
        >
          Portugal and 6 more countries have just been added to the UK travel green list
        </p>
        <button
          class="bpk-button bpk-button--primary-on-dark bpk-graphic-promo__cta"
          tabindex="-1"
          type="button"
        >
          Learn more
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BpkGraphicPromo should support custom class names 1`] = `
<DocumentFragment>
  <div
    aria-label="Sponsored. Airline Name. Ride your wave. Portugal and 6 more countries have just been added to the UK travel green list. Learn more"
    class="bpk-card bpk-graphic-promo custom-classname"
    id="mock-content-id"
    role="link"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="bpk-graphic-promo__container bpk-graphic-promo__container--start"
      id="mock-content-id__content"
    >
      <div
        class="bpk-graphic-promo__sponsor-content bpk-graphic-promo__sponsor-content--start"
      >
        <span
          class="bpk-text bpk-text--body-default bpk-graphic-promo__sponsor-label"
        >
          Sponsored
        </span>
        <img
          alt="Airline Name"
          class="bpk-graphic-promo__sponsor-logo"
          src="path/to/logo.png"
        />
      </div>
      <div
        class="bpk-graphic-promo__promo-content bpk-graphic-promo__promo-content--start"
      >
        <h2
          class="bpk-text bpk-text--body-default bpk-graphic-promo__headline"
        >
          Ride your wave
        </h2>
        <p
          class="bpk-text bpk-text--body-default bpk-graphic-promo__subheading"
        >
          Portugal and 6 more countries have just been added to the UK travel green list
        </p>
        <button
          class="bpk-button bpk-button--primary-on-dark bpk-graphic-promo__cta"
          tabindex="-1"
          type="button"
        >
          Learn more
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
